Widen integer type compared to enum

GCC 10 seems to enforce signedness between types when comparing values.

Author: Lars Viklund <zao@zao.se>
diff -ru binutils-2.31.1.orig/gold/x86_64.cc binutils-2.31.1/gold/x86_64.cc
--- binutils-2.31.1.orig/gold/x86_64.cc	2018-06-24 20:38:57.000000000 +0200
+++ binutils-2.31.1/gold/x86_64.cc	2020-05-13 00:56:30.254725766 +0200
@@ -1585,7 +1585,7 @@
 {
   uint32_t val = 0;
 
-  switch (pr_type)
+  switch ((size_t)(uint32_t)pr_type)
     {
     case elfcpp::GNU_PROPERTY_X86_ISA_1_USED:
     case elfcpp::GNU_PROPERTY_X86_ISA_1_NEEDED:
@@ -1606,7 +1606,7 @@
       break;
     }
 
-  switch (pr_type)
+  switch ((size_t)(uint32_t)pr_type)
     {
     case elfcpp::GNU_PROPERTY_X86_ISA_1_USED:
       this->isa_1_used_ |= val;
